uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Form/CreateUserRecordTaskForm.php

```diff
@@ -44,7 +44,14 @@ public function postProcess() {
     $contactsToCreate = $this->getValidContactsForCreation();
 
     foreach ($contactsToCreate as $contact) {
-      $this->createAccount($contact, $roles);
+      $user = $this->drupalUserService->findUserByEmail($contact['email']);
+      if (!$user) {
+        $this->createAccount($contact, $roles);
+      }
+      else {
+        $this->drupalUserService->processUserAccount($user, $roles);
+        $this->addUfId($user, $contact['id']);
```

Comment:
I don't think we need to add the `processUserAccount` method to the DrupalUserService class. For restoring accounts, what is basically happening is just restoring the user roles. I would suggest the following:
- Revert changes to the `createNew` method. 
- Change the `findUserByEmail` to something like `userExistsForEmail` that returns a boolean
- Use the Drupal user service `addRoles` method to add the roles the contact email if a user account for that email already exists, if not create a new user account(default behaviour)
- Extract the function to addUserRole and addUfId to a new function `restoreAccount` within the form class for clarity.

This way we would only need just one public method to the DrupalUserService class and not two. 
Then please add a test for the `userExistsForEmail`(or a better name) method, see sample tests that uses drupal mock function in for the Drupal service test classes.

Reply:
Thanks. I will refactor the implementation.